From e496d24153b42230c2de4b8ec968a25181eb44bb Mon Sep 17 00:00:00 2001
From: chongbiao Chen <chenchongbiao@radxa.com>
Date: Wed, 3 Dec 2025 13:52:01 +0800
Subject: [PATCH 1/1] net: phy: qca808x: add support for radxa-standard-led
 device tree property

Add support for the 'radxa-standard-led' boolean property in the PHY.the driver configures LED0 (yellow) to be constantly on for all link speeds (10/100/1000 Mbps), while leaving LED1 (green) to its default activity blinking behavior.

Signed-off-by: chenchongbiao <chenchongbiao@radxa.com>
---
 .../dts/qcom/qcs9075-radxa-airbox-q900.dts    | 32 +++++++++++++++++++
 drivers/net/phy/qcom/Kconfig                  |  1 +
 2 files changed, 33 insertions(+)

diff --git a/src/arch/arm64/boot/dts/qcom/qcs9075-radxa-airbox-q900.dts b/src/arch/arm64/boot/dts/qcom/qcs9075-radxa-airbox-q900.dts
index 0a45b15f1..17acd828e 100644
--- a/src/arch/arm64/boot/dts/qcom/qcs9075-radxa-airbox-q900.dts
+++ b/src/arch/arm64/boot/dts/qcom/qcs9075-radxa-airbox-q900.dts
@@ -401,6 +401,22 @@ sgmii_phy0: phy@1c {
 			reset-gpios = <&pmm8654au_2_gpios 8 GPIO_ACTIVE_LOW>;
 			reset-assert-us = <11000>;
 			reset-deassert-us = <70000>;
+
+			leds {
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				led@0 {
+					reg = <0>;
+					active-low;
+					linux,default-trigger = "default-on";
+				};
+
+				led@1 {
+					reg = <1>;
+					active-low;
+				};
+			};
 		};
 	};
 
@@ -489,6 +505,22 @@ sgmii_phy1: phy@1c {
 			reset-gpios = <&pmm8654au_2_gpios 9 GPIO_ACTIVE_LOW>;
 			reset-assert-us = <11000>;
 			reset-deassert-us = <70000>;
+
+			leds {
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				led@0 {
+					reg = <0>;
+					active-low;
+					linux,default-trigger = "default-on";
+				};
+
+				led@1 {
+					reg = <1>;
+					active-low;
+				};
+			};
 		};
 	};
 
diff --git a/src/drivers/net/phy/qcom/Kconfig b/src/drivers/net/phy/qcom/Kconfig
index 80db24deb..69c3e6fee 100644
--- a/src/drivers/net/phy/qcom/Kconfig
+++ b/src/drivers/net/phy/qcom/Kconfig
@@ -17,6 +17,7 @@ config QCA83XX_PHY
 
 config QCA808X_PHY
 	tristate "Qualcomm QCA808x PHYs"
+	default m
 	select QCOM_NET_PHYLIB
 	help
 	  Currently supports the QCA8081 model
-- 
2.43.0

